include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.td

```diff
@@ -262,20 +267,20 @@ def FModuleLike : OpInterface<"FModuleLike"> {
       return "portSyms";
     }
 
-    /// Replace NUll entries with invalid sym and clear array if all elements
+    /// Replace NULL entries with invalid sym and clear array if all elements
     /// are invalid.
     static void fixupPortSymsArray(SmallVectorImpl<Attribute> &syms,
                                    MLIRContext *context) {
-      bool isValid = false;
-      for (size_t i = 0, e = syms.size(); i < e; ++i)
-        if (!syms[i])
-          syms[i] = InnerSymAttr::get(context);
-        else
-          isValid = true;
       // The lack of *any* port symbol is represented by an empty `portSyms`
       // array as a shorthand.
-      if (!isValid)
+      if (llvm::none_of(syms, [&](auto &sym) { return sym; })) {
         syms.clear();
+        return;
+      }
+      auto empty = InnerSymAttr::get(context);
```

Comment:
This  is a nice catch, I didn't realize the savings of caching the empty Attribute. Must remember this.

Reply:
To be fair, we were operating on modules with thousands of ports.  But apparently, yes!  Good to know, agreed, I saw it on profile stack but also in MLIR's SymbolTable .  Good to keep in mind, agreed!